firmware/components/driver_framebuffer/include/driver_framebuffer_matrix_drawing.h

```diff
@@ -0,0 +1,58 @@
+
+#pragma once
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef union rgb_splitter_union_t {			// Used to seperate ARGB more efficiently due to the way CPUs work
+	struct {
+		uint8_t alpha;
+		uint8_t red;
+		uint8_t green;
+		uint8_t blue;
+	} seperate;
+	uint32_t together;
+} rgb_splitter_union;
+
+typedef struct texture_2d_t {
+	uint32_t *buffer;
+	int16_t width;
+	int16_t height;
+    bool interpolation;
+} texture_2d;
+
+//shader_2d(float u, float v, int16_t x, int16_t y, void *args)
+typedef uint32_t(*shader_2d)(float, float, int16_t, int16_t, void*);
+
+typedef struct triangle_uv_t {
+	float u0, v0;
+	float u1, v1;
+	float u2, v2;
+} triangle_uv;
+
+void driver_framebuffer_triangle(Window* window, float x0, float y0, float x1, float y1, float x2, float y2, uint32_t color);
+/* Draws a triangle. */
```

Comment:
I would put descriptive comments above code instead of below, as that is common practice in programming

Reply:
Function name explains that it draws a triangle, not very usefull to have this comment here at all.